### PR TITLE
Fix showing spinners on the homepage

### LINKED ISSUE #31503: Spinner appears on the top left on the sidebar
### Describe the bug

We make the spinner appear on a weird place when loading the sidebar, should be vertically and horizontally aligned
image

### To Reproduce

1) open the browser dev tools, go to network and switch the connection to fast 4g
2) enter the internal metabase instance
3) refresh, see the spinner in the wrong place

### Expected behavior

vertically and horizontally aligned?

### Logs

_No response_

### Information about your Metabase installation

```JSON
version: master
```

### Severity

P3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/31503

Changes:
- Fixes spinner layout
- Does not load or wait for metabot-related data when metabot is not enabled

How to verify:
- Throttle network in dev tools
- Refresh the home page

Screenshot 2023-06-28 at 22 09 03

### CODE REVIEW

File: frontend/src/metabase/home/components/HomePage/HomePage.tsx [hunk 1]
```diff
@@ -26,37 +25,71 @@ import { HomeContent } from "../HomeContent";
 const SEARCH_QUERY = { models: "dataset", limit: 1 } as const;
 
 export const HomePage = (): JSX.Element => {
-  const databaseListState = useDatabaseListQuery();
```
Comment (ranquild) [inline frontend/src/metabase/home/components/HomePage/HomePage.tsx:29]: Cleans up this by moving to separate hooks

File: frontend/src/metabase/home/components/HomePage/HomePage.tsx [hunk 2]
```diff
@@ -26,37 +25,71 @@ import { HomeContent } from "../HomeContent";
...
     if (!isSmallScreen()) {
       dispatch(openNavbar());
     }
   }, [dispatch]);
+};
+
+const useMetabot = () => {
+  const isMetabotEnabled = useSelector(getIsMetabotEnabled);
+  const databaseListQuery = useDatabaseListQuery({
+    enabled: isMetabotEnabled,
```
Comment (ranquild) [inline frontend/src/metabase/home/components/HomePage/HomePage.tsx:57]: Only load data when metabot is enabled

File: frontend/src/metabase-types/api/settings.ts [hunk 3]
```diff
@@ -180,7 +180,7 @@ export interface Settings {
   "custom-homepage": boolean;
   "custom-homepage-dashboard": number | null;
   "deprecation-notice-version"?: string;
-  "dismissed-custom-dashboard-toast": boolean;
+  "dismissed-custom-dashboard-toast"?: boolean;
```
Comment (ranquild) [inline frontend/src/metabase-types/api/settings.ts:183]: The setting is not present on the page load, it also gets loaded via `/api/session/properties`. This makes a difference for the home page

File: frontend/src/metabase/home/components/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -26,37 +26,72 @@ import { HomeContent } from "../HomeContent";
...
+  return {
+    databases: databaseListQuery.data ?? [],
+    models: searchListQuery.data ?? [],
+    isMetabotEnabled,
+    isLoading: databaseListQuery.isLoading || searchListQuery.isLoading,
+    error: databaseListQuery.error ?? searchListQuery.error,
+  };
+};
+
+const getHasMetabot = (
```
Comment (uladzimirdev) [inline frontend/src/metabase/home/components/HomePage/HomePage.tsx:74]: confusing name, can it be `isMetabotSupported`?

File: frontend/src/metabase/home/components/HomePage/HomePage.tsx [hunk 5]
```diff
@@ -26,37 +26,72 @@ import { HomeContent } from "../HomeContent";
...
+  isMetabotEnabled: boolean,
+) => {
+  const hasModels = models.length > 0;
+  const hasSupportedDatabases = databases.some(canUseMetabotOnDatabase);
+  return hasModels && hasSupportedDatabases && isMetabotEnabled;
+};
+
+const useDashboardPage = () => {
+  const dashboardId = useSelector(getCustomHomePageDashboardId);
+  const isLoading = useSelector(getSettingsLoading);
```
Comment (uladzimirdev) [inline frontend/src/metabase/home/components/HomePage/HomePage.tsx:86]: nit: isSettingsLoading

File: frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.styled.tsx [hunk 6]
```diff
@@ -115,6 +115,16 @@ export const CollectionMenuList = styled.ul`
 `;
 
 export const LoadingContainer = styled.div`
+  display: flex;
+  flex: 1;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const LoadingContent = styled.div`
```
Comment (uladzimirdev) [inline frontend/src/metabase/nav/containers/MainNavbar/MainNavbar.styled.tsx:124]: can't it be Box?

**Review by uladzimirdev — APPROVED** [review]

**Pull request merged**